avocado/utils/software_manager.py

```diff
@@ -658,6 +767,62 @@ def provides(self, name):
         except process.CmdError:
             return None
 
+    def source_install(self, name):
```

Comment:
Do you see this to be frequently used? It's tough call but I think it could be private function only (`_source_install`).